Automattic/WP-Job-Manager
# Add settings UI for email notifications

Fixes #1439 

Example of future settings:
* For #446 we'll add a setting for how many days ahead we should warn employers before their listings expire.
* We could pretty easily make the subject customizable.
* For some notifications, the entire notification could be customizable. 

#### Changes proposed in this Pull Request:

* Restructures `WP_Job_Manager_Settings` (318c1405663e7de250588264296e616b0ccbc967) to support the new field type `multi` (compound settings stored in serialized arrays) and `mutli_enable_expand`.
* Adds basic fields for enabled and send in plain/rich text.
* Does some minor UI change to settings (adds line separator) to help a bit with the new email notifications screen.

**Round 2 Design**:
![email-notifications-demo](https://user-images.githubusercontent.com/68693/38770246-ce3bd1e8-4007-11e8-964c-364125cbb73f.gif)
<img width="1336" alt="screen shot 2018-04-14 at 5 19 49 pm" src="https://user-images.githubusercontent.com/68693/38777367-9ab35232-409e-11e8-9cf0-67c3665c46a7.png">

<a href="https://user-images.githubusercontent.com/68693/38757143-d89fff5a-3f63-11e8-98b5-a36ea6b5adda.png">Round 1 Design</a>

#### Testing instructions:

* Ensure all tests pass.
* Try enabling/disabling these settings and make sure the notifications are sent when appropriate in the right type (rich/plain).

cc: @tripflex 

To show what it looks like when we add another setting:
<img width="1322" alt="screen shot 2018-04-15 at 2 30 03 pm" src="https://user-images.githubusercontent.com/68693/38778974-93bc16b0-40b9-11e8-939b-5cf4b4af28e0.png">
